packages/vue-app/template/utils.js

```diff
@@ -25,6 +25,16 @@ export function interopDefault (promise) {
 export function hasFetch(vm) {
   return vm.$options && typeof vm.$options.fetch === 'function' && !vm.$options.fetch.length
 }
+export function purifyData(data) {
```

Comment:
Since this adds a runtime overhead, maybe we can warn users in development mode to not use non POJO in state/data?

Reply:
I think a lot of people don't understand the POJO warning - and indeed the side effects (that the entire component state is stringified, for example!)

Are you thinking of warning in dev mode and not checking in prod?

Another option: use a Proxy for `this` for fetch to detect changed properties so not everything will be stringified?

Reply:
> I think a lot of people don't understand the POJO warning - and indeed the side effects (that the entire component data is stringified, for example!)

Indeed fully agree. We need better DX to guide users about this.

> Are you thinking of warning in dev mode and not checking in prod?

Yeah. Like a `process.env` guard similar to other vue packages (vuex for example only checks `stat` to be a function in dev env)  this can avoid adding overhead to production

> Another option: use a Proxy for this for fetch to detect changed properties so not everything will be stringified?

Nuxt 3 PR welcome :D